inc/entity.class.php

```diff
@@ -3415,7 +3457,65 @@ static function getIcon() {
       return "fas fa-layer-group";
    }
 
+   /**
+    * Get values for contracts_id_default field
+    *
+    * @since 9.5
+    *
+    * @return array
+    */
+   public static function getDefaultContractValues($entities_id): array {
+      $values = [
+         self::CONFIG_PARENT => __('Inheritance of the parent entity'),
+         -1 => __('Contract in ticket entity'),
+      ];
+
+      $contract = new Contract();
+      $criteria = getEntitiesRestrictCriteria('', '', $entities_id, true);
+      $criteria[] = Contract::getExpiredCriteria();
+      $contracts = $contract->find($criteria);
+
+      foreach ($contracts as $contract) {
+         $values[$contract['id']] = $contract['name'];
+      }
+
+      return $values;
+   }
+
    public static function getAnonymizeConfig(?int $entities_id = null) {
       return Entity::getUsedConfig('anonymize_support_agents', $entities_id);
    }
+
+   public static function getDefaultContract(int $entities_id): int {
+      // Entity isn't set, can't load config
+      if ($entities_id === false) {
+         return 0;
+      }
```

Comment:
Due to `$entities_id` param type hinting, it is not possible to have a `false` value here.
```suggestion
```